### DISCUSSION TITLE
Episode 201: CppCon 2024 Recap!

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#100] [Episode]
Episode Link: https://adspthepodcast.com/2024/09/27/Episode-201.html

Feel free to write comments or ask questions 🙂

**Comment by VincentZalzal** [comment]
I am excited by having Ben Deane as co-host: very knowledgeable and effective communicator. Looking forward to more podcasts with you two!

I haven't watched Ben's boolean algebra talk (not available online at the time of writing). I'm not sure it is relevant, but I too am quite fond of logical implication in code, especially in asserts. I remember adding special assert imply macros in a previous codebase I worked on, but it was not a popular feature among my coworkers, making me doubt about their clarity in the end.

To give an example: let's say you're implementing `find_if` and want to assert its postconditions. One of them is: if `find_if` returns a valid iterator, then the predicate should return true on its value. With normal asserts, this looks like this:
```cpp
assert(it == last || pred(*it));
```
Notice how I have to negate the first part of the english sentence to express the assertion. I have to convert A ⇒ B into ¬A ∨ B, which I feel is less expressive and error-prone.
